reference/7.1/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md

```diff
@@ -122,6 +122,123 @@ Param(
 )
 ```
 
+## Switch parameters
+
+Switch parameters are parameters with no parameter value. They're effective
+only when they're used and have only one effect.
+
+For example, the **NoProfile** parameter of **pwsh.exe** is a switch
+parameter.
+
+To create a switch parameter in a function, specify the `Switch` type in the
+parameter definition.
+
+For example:
+
+```powershell
+Param([Switch]<ParameterName>)
+```
+
+Or, you can use an another method:
+
+```powershell
+Param(
+    [Parameter(Mandatory=$false)]
+    [Switch]
+    $<ParameterName>
+)
+```
+
+Switch parameters are easy to use and are preferred over Boolean parameters,
+which have a more difficult syntax.
+
+For example, to use a switch parameter, the user types the parameter in the
+command.
+
+`-IncludeAll`
+
+To use a Boolean parameter, the user types the parameter and a Boolean value.
+
+`-IncludeAll $true`
+
+When creating switch parameters, choose the parameter name carefully. Be sure
+that the parameter name communicates the effect of the parameter to the user.
+Avoid ambiguous terms, such as **Filter** or **Maximum** that might imply a
+value is required.
+
+## Dynamic parameters
+
+Dynamic parameters are parameters of a cmdlet, function, or script that are
+available only under certain conditions.
+
+For example, several provider cmdlets have parameters that are available only
+when the cmdlet is used in the provider drive, or in a particular path of the
+provider drive. For example, the **Encoding** parameter is available on the
+`Add-Content`, `Get-Content`, and `Set-Content` cmdlets only when it's used in
+a file system drive.
+
+You can also create a parameter that appears only when another parameter is
+used in the function command or when another parameter has a certain value.
+
+Dynamic parameters can be useful, but use them only when necessary, because
+they can be difficult for users to discover. To find a dynamic parameter, the
+user must be in the provider path, use the **ArgumentList** parameter of the
```

Comment:
I would rewrite this paragraph to be something more like as follows:

> Dynamic parameters can be useful, but because they are difficult to discover they should only be implemented when no other options (such as parameter sets) are applicable. In particular, dynamic parameters can depend on the currently active PowerShell provider and require additional information to appear in discovery mechanisms:
>   - With `Get-Command`, the `-ArgumentList` parameter must be used to present the arguments that would be passed to the command in order to discover the dynamic parameters that might appear.
>   - With `Get-Help`, the `-Path` parameter allows a provider path to be specified to add provider-specific dynamic parameters to the help output.